### PR TITLE
T132812694, T132811023, T132809714: check for //# sourceURL alert case, math tags, svg use tags

### PR DESCRIPTION
**Pull request by agango**

### CODE REVIEW

File: src/js/contentUtils.js [hunk 1]
```diff
@@ -594,6 +594,16 @@ async function processJSWithSrc(script, origin, version) {
...
+      // check to ensure there are no popups via MITM attack after //# sourceURL - check string contents after sourceURLIndex
+      const afterSourceURL = sourceText.slice(
+        sourceURLIndex,
+        sourceText.length
+      );
+      if (afterSourceURL.includes('alert')) {
+        return {
+          valid: false,
+        };
+      }
```
Comment (rich-hansen) [inline src/js/contentUtils.js:606]: We chatted about this over work chat. We need to remove this and beef up the detection

**Review by rich-hansen — CHANGES_REQUESTED** [review]

File: src/js/contentUtils.js (hunk 1 above)
Comment (agango) [reply src/js/contentUtils.js]: added more url detection, including checking to ensure that the source is coming from fbcdn. Are there any source files that may not come from "static.xx.fbcdn.net"?

File: src/js/contentUtils.js [hunk 2]
```diff
@@ -484,12 +486,38 @@ export function hasInvalidAttributes(htmlElement) {
...
+        Object.keys(childNode.attributes).length >= 1
+      ) {
+        Array.from(childNode.attributes).forEach(elementAttribute => {
+          if (
+            (elementAttribute.localName === 'href' ||
+              elementAttribute.localName === 'xlink:href') &&
+            childNode
+              .getAttribute(elementAttribute.localName)
+              .toLowerCase()
+              .includes('javascript')
```
Comment (rich-hansen) [inline src/js/contentUtils.js:507]: does it need to be includes or can it be `startsWith`?

File: src/js/contentUtils.js [hunk 3]
```diff
@@ -572,6 +600,19 @@ export const scanForScripts = () => {
...
 };
 
+function checkForUrl(source) {
+  // the source URL has the following format: '//# sourceURL={url}', so we can look for '=' and check for the url from that index + 1
+  const urlIndex = source.indexOf('=') + 1;
+  if (urlIndex.slice(0, 4) !== 'http' && urlIndex.slice(0, 5) !== 'https') {
+    return false;
+  }
+  // check to ensure source is being sent from fbcdn
+  else if (!urlIndex.includes('static.xx.fbcdn.net')) {
```
Comment (rich-hansen) [inline src/js/contentUtils.js:610]: Don't need this check

**Review by rich-hansen — CHANGES_REQUESTED** [review]

File: src/js/contentUtils.js (hunk 2 above)
Comment (agango) [reply src/js/contentUtils.js:507]: yeah, we can make it startsWith

**Review by rich-hansen — APPROVED** [review]

**Pull request merged**